ivy/functional/backends/tensorflow/experimental/sorting.py

```diff
@@ -11,3 +11,29 @@ def msort(
     out: Optional[Union[tf.Tensor, tf.Variable]] = None,
 ) -> Union[tf.Tensor, tf.Variable]:
     return tf.sort(a, axis=0)
+
+
+# lexsort
+def lexsort(
+    keys: Union[tf.Tensor, tf.Variable],
+    /,
+    *,
+    axis: int = -1,
+    out: Optional[Union[tf.Tensor, tf.Variable]] = None,
+) -> Union[tf.Tensor, tf.Variable]:
+    shape = keys.shape.as_list()
+    if len(shape) == 1:
+        return tf.argsort(keys, axis=axis, stable=True)
+    if shape[0] == 0:
+        raise TypeError('need sequence of keys with len > 0 in lexsort')
+    if len(shape) == 2 and shape[1] == 1:
+        return tf.convert_to_tensor([0])
+    result = tf.argsort(keys[0], axis=axis, stable=True)
+    if shape[0] == 1:
+        return result
+    for i in range(1, shape[0]):
+        key = keys[i]
+        ind = tf.gather(key, result)
+        temp = tf.argsort(ind, axis=axis, stable=True)
+        result = tf.gather(result, temp)
+    return tf.cast(result, tf.int64)
```

Comment:
here I cast all outputs to be int64 just like what numpy will do, but according to the test(https://github.com/unifyai/ivy/actions/runs/4137025627/jobs/7151674282), it still somehow returns int32. How is this possible? @VedPatwardhan

Reply:
Actually, there's a number of return statements before this one, and the casting is not applied to those (e.g. line 26, 30, 33). Also, I'd suggest having the result with the same dtypes as the input, rather than flat out type casting to `int64` 🙂

Reply:
Well, this function takes keys as input and always return integer indexes that can be used to sort the given keys, because of this, the return type will always be int64 as it represents position and is not influenced by the origenal type of the inputs but only their relative differences.